bin/uglifyjs

```diff
@@ -254,7 +258,17 @@ if (specified["self"]) {
     if (!options.wrap) options.wrap = "UglifyJS";
     paths = UglifyJS.FILES;
 }
-if (paths.length) {
+if (specified["in-situ"]) {
+    if (specified["reduce-test"] || specified["self"]) fatal("incompatible options specified");
+    paths.forEach(function(name) {
+        print(name);
+        if (/^ast|spidermonkey$/.test(name)) fatal("invalid file name specified");
+        files = {};
+        files[convert_path(name)] = read_file(name);
+        output = name;
+        run();
+    });
```

Comment:
>There's so little code changed in `bin/uglifyjs` to implement `--in-situ` that I don't know how it works.

All the parsing and conversion of CLI into API options are done above, and below here its just getting the input code either from file(s) or STDIN &minus; just the right place to add a third mode of operation here 😎

Reply:
> Perhaps the shell function `minify_in_situ` can be dropped altogether:

I kinda like the ability to centrailise the bisection modification though.

Reply:
What happens in this scenario? Does it stop at the first bad input and what would the exit code be?
```
bin/uglifyjs -mc --in-situ good1.js bad1.js good2.js bad2.js good3.js
```
And what happens if, let's say, `good2.js` is read-only and the file cannot be rewritten?

Reply:
It will just be whatever we print out and rage-quit with `bin/uglifyjs -mc bad1.js`, with an extra line of `good1.js` printed beforehand.

Process exit code will be non-zero.

Reply:
> And what happens if, let's say, `good2.js` is read-only and the file cannot be rewritten?

Then it will die here:
https://github.com/mishoo/UglifyJS/blob/01fa430a3e20098361ff8d35306ba1c31fea514c/bin/uglifyjs#L400-L402

Reply:
I see. I didn't try this PR. Does each file name get emitted to stdout as it goes along?

Reply:
For my rMBP it looks the same, but seems like I/O is quite a bit more of a bottleneck with macOS on bare metal than say Ubuntu on GitHub Actions.

Reply:
Do we need to bother with mocha tests for the error scenarios above?  Or just file it under "internal debug use only"?

Reply:
Given that we mostly reuse non-debug code here, I think existing coverage under `mocha/cli.js` should be good enough.